File: src/models/Fare.js
const mongoose = require("mongoose");

const fareSchema = new mongoose.Schema(
    {
        rideId: {
            type: String,
            required: true
        },

        distance: {
            type: Number,
            required: true
        },

        baseFare: {
            type: Number,
            required: true
        },

        perKmRate: {
            type: Number,
            required: true
        },

        totalFare: {
            type: Number,
            required: true
        }
    },
    {
        timestamps: true
    }
);

module.exports = mongoose.model("Fare", fareSchema);

File: fare-payment-service/src/main/java/com/ridelink/farepayment/model/Fare.java
package com.ridelink.farepayment.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "fares")
public class Fare {

    @Id
    private String id;

    private String rideId;
    private double distance;
    private double baseFare;
    private double ratePerKm;
    private double totalFare;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getRideId() {
        return rideId;
    }

    public void setRideId(String rideId) {
        this.rideId = rideId;
    }

    public double getDistance() {
        return distance;
    }

    public void setDistance(double distance) {
        this.distance = distance;
    }

    public double getBaseFare() {
        return baseFare;
    }

    public void setBaseFare(double baseFare) {
        this.baseFare = baseFare;
    }

    public double getRatePerKm() {
        return ratePerKm;
    }

    public void setRatePerKm(double ratePerKm) {
        this.ratePerKm = ratePerKm;
    }

    public double getTotalFare() {
        return totalFare;
    }

    public void setTotalFare(double totalFare) {
        this.totalFare = totalFare;
    }
}